### PR TITLE
Update JsonSerializer exception

### PR DESCRIPTION
**Pull request by gewarren**

### CODE REVIEW

File: xml/System.Text.Json/JsonSerializer.xml [hunk 1]
```diff
@@ -307,7 +307,7 @@ For more information, see [How to serialize and deserialize JSON](/dotnet/standa
 ]]></format>
         </remarks>
         <exception cref="T:System.IndexOutOfRangeException">
-          .NET Core 3.0 only: <typeparamref name="TValue" /> is <see cref="T:System.Char" /> and <paramref name="json" /> is an empty string.</exception>
+          .NET Core 3.0 and earlier versions only: <typeparamref name="TValue" /> is <see cref="T:System.Char" /> and <paramref name="json" /> is an empty string.</exception>
```
Comment (ahsonkhan) [inline xml/System.Text.Json/JsonSerializer.xml:310]:
```suggestion
          Only .NET Core 3.0 and older stand-alone package version (4.6.0): <typeparamref name="TValue" /> is <see cref="T:System.Char" /> and <paramref name="json" /> is an empty string.</exception>
```

File: xml/System.Text.Json/JsonSerializer.xml [hunk 2]
```diff
@@ -307,7 +307,7 @@ For more information, see [How to serialize and deserialize JSON](/dotnet/standa
 ]]></format>
         </remarks>
         <exception cref="T:System.IndexOutOfRangeException">
```
Comment (ahsonkhan) [inline xml/System.Text.Json/JsonSerializer.xml:309]: Looking at the whole docs experience again, I don' think these version specific exceptions should be in this xml file.

We don't need (nor want) folks to see the Deserialize method can throw IOOR, **especially not on 3.1 and 5.0 (and beyond)**
That was a point-in-time bug in the code for a super uncommon edge case that folks shouldn't run into anyway.

This exception isn't something we want users to deal with, especially going forward.

**Please remove this exception all together.**

The docs are already versioned. A reader will ask, why am I seeing exceptions for the version I am not even using:
image

**Review by ahsonkhan — CHANGES_REQUESTED** [review]
Remove the IOOR exception from the docs all together. We shouldn't encode a point-in-time, version specific, buggy behavior into the public facing docs.

**Comment by ahsonkhan** [comment]
Same with the other note:
https://github.com/dotnet/dotnet-api-docs/pull/4953#discussion_r501964734

File: xml/System.Text.Json/JsonSerializer.xml (hunk 2 above)
Comment (gewarren) [reply xml/System.Text.Json/JsonSerializer.xml:309]: Unfortunately the exceptions and remarks aren't versioned, only the API signature. So this is the way we document version-specific information. But I will remove it.

**Comment by ahsonkhan** [comment]
I was going to ask, for a clean change, if we could just revert the whole change from https://github.com/dotnet/dotnet-api-docs/pull/4953

```bash
git revert <commit hash from previous PR>
```

But it looks like we don't mandate using `Squash and merge` for the docs repo to make sure there is only a single commit for each PR. @mairaw, @tdykstra - what's the reason for that? Almost all the other dev repos I have worked in, we almost always "Squash and merge", so much so, that we have disabled doing anything else by default.

https://github.blog/2016-04-01-squash-your-commits/#whats-changing

**Review by ahsonkhan — APPROVED** [review]

**Comment by mairaw** [comment]
@ahsonkhan yes I agree that squash and merge is the best approach. The reason why we can’t disable the regular merge is because when we merge to live, we can’t squash the commits.

**Comment by ahsonkhan** [comment]
> because when we merge to live, we can’t squash the commits.

And why is that?

**Comment by mairaw** [comment]
Because we need to keep the commits intact when we merge one branch to the other, otherwise the branches don’t match.

**Comment by ahsonkhan** [comment]
If the goal is to keep the master and live branches in-sync, would it be possible to automate that through per-commit "sync" PRs, or even go around PRs and port the changes directly to the branch?

**Pull request merged**